scripts/postgresql-setup.sh

```diff
@@ -145,13 +153,14 @@ function dump_schema {
 
 function create_snapshot {
 	tgz_file=$1.tgz
-	dbfile=$1.sql
 	ledger_file=$2
 	tmp_dir=$(mktemp "${directory}" -t db-sync-snapshot-XXXXXXXXXX)
 	echo $"Working directory: ${tmp_dir}"
-	pg_dump --no-owner --schema=public "${PGDATABASE}" > "${tmp_dir}/$1.sql"
-	cp "$ledger_file" "$tmp_dir/$(basename "${ledger_file}")"
-	tar zcvf - --directory "${tmp_dir}" "${dbfile}" "$(basename "${ledger_file}")" | tee "${tgz_file}.tmp" \
+	pg_dump --no-owner --schema=public --jobs="${numcores}" "${PGDATABASE}" --format=directory --file="${tmp_dir}/db/"
+	lstate_gz_file=$(basename "${ledger_file}").gz
+	gzip --to-stdout "$ledger_file" > "$tmp_dir/$(basename "${ledger_file}").gz"
+	# Use plain tar here because the database dump files and the ledger state file are already gzipped.
+	tar cvf - --directory "${tmp_dir}" "db" "$(basename "${ledger_file}")" | tee "${tgz_file}.tmp" \
```

Comment:
Also should this not be ( to add the .gz )?:

tar cvf - --directory "${tmp_dir}" "db" "$(basename "${ledger_file}").gz" | tee "${tgz_file}.tmp" \

Reply:
Still debugging this. Only pushed to the repo because I needed to pull it on another machine.